### ISSUE TITLE
Issue to use "input()" with the Brython editor

### ISSUE THREAD

**Issue by pierrewtLanca** [brython-dev/brython#1908]
Hi, 

Sorry if this bug is already reported.

Despite the Brython editor being very useful, I think it has a major problem in the way it handles the Python input() function (well, that might not just be the editor actually, but that's what I used). In fact,t the problem is not about "input()" but it's the way the editor works in general.

I expect the input to prompt the user and **return to to the main flow** right after. However, with the JS prompt, the editor doesn't get anything until the whole code has been run - which is of course a very big issue.
For example, in the editor, use this code:
```
testVar = ""
while(testVar != "stop"):
    testVar = input("type a number to print it, or 'stop' to stop")
    print("you typed " + testVar)
print("good bye")
```
What is expected here is a continuous prompting / displaying the input until "stop" is typed. But it will keep prompting without showing anything in the editor until the "end" of the script. In other words, the editor isn't running "on the fly".

I don't know if the reason for that is due to the conversion from Python to JS but that is a really issue in term of workflow.

The problem is not just because of "input()" but is apparent here. If a simple code do a infinite while loop printing something, the editor won't show anything because it waits for the script to "finish". 

Is there a way we can get the editor running "on the fly"? 

Thanks a lot,
Pierre

**Comment by pierrewtLanca** [comment]
Ah, I just saw that other issues were raised before, sorry.. https://github.com/aroberge/reeborg/issues/289
Well I guess the question is now, is there any update about this issue?
Thanks a lot,
Pierre

**Comment by PierreQuentel** [comment]
Unfortunately no update, until Javascript supports user-defined blocking functions, which probably means never :-(

You have to use workarounds like
```python
testVar = ""
while testVar != 'stop':
    if not testVar:
        testVar = input("Type a number to print it, or 'stop' to stop")
    else:
        testVar = input("You typed " + testVar + "\nPlease try again")
print("good bye")
```

**Comment by pierrewtLanca** [comment]
Hi
Thanks for your answer. 
The program I want to run needs to display what the input was, right when the input was validated from the prompt.
Actually, we noticed that Chrome was the problem of this issue: Firefox will show the output of our Python "print()" right when it is called (in the Brython editor).
So we can at least rely on this behaviour for the moment, but I don't really get why with Chrome we only get the output once we have passed the loop.
Thanks,
Pierre
